### ISSUE TITLE
Avocado instrumented timeout in setUp and tearDown error

### ISSUE THREAD

**Issue by richtja** [avocado-framework/avocado#6013]
**Describe the bug**
When the test timeout is reached during `setUp` or `tearDown` methods, the test results as `ERROR` instead of `INTERUPTED`. Based on my testing, the interruption is handled properly, only the result value is wrong.

**Steps to reproduce**
Steps to reproduce the behavior with a reproducible whenever possible.
Test setUp example:
```
import time

from avocado import Test


class TimeoutTest(Test):
    """
    Functional test for avocado. Throw a TestTimeoutError.

    :param sleep_time: How long should the test sleep
    """

    timeout = 3

    def setUp(self):
        time.sleep(5)

    def test(self):
        """
        This should throw a TestTimeoutError.
        """
        pass
```
Test tearDown example:
```
import time

from avocado import Test


class TimeoutTest(Test):
    """
    Functional test for avocado. Throw a TestTimeoutError.

    :param sleep_time: How long should the test sleep
    """

    timeout = 3

    def test(self):
        """
        This should throw a TestTimeoutError.
        """
        pass

    def tearDown(self):
        time.sleep(5)
```

**Expected behavior**
```
$ avocado run test.py
JOB ID     : d3e279280bb6666bc0c4b05fdcbd5d8b2e4ff135
JOB LOG    : /home/janrichter/avocado/job-results/job-2024-08-28T10.12-d3e2792/job.log
 (1/1) test.py:TimeoutTest.test: STARTED
 (1/1) test.py:TimeoutTest.test: INTERRUPTED: Test interrupted: Timeout reached (3.07 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 1 | CANCEL 0
JOB HTML   : /home/janrichter/avocado/job-results/job-2024-08-28T10.12-d3e2792/results.html
JOB TIME   : 4.79 s
```

**Current behavior**
What you are getting?
```
$ avocado run test.py
JOB ID     : f11768c3dfd8f2356fdd94b79300a84c167ac717
JOB LOG    : /home/janrichter/avocado/job-results/job-2024-08-28T10.10-f11768c/job.log
 (1/1) test.py:TimeoutTest.test: STARTED
 (1/1) test.py:TimeoutTest.test: ERROR: Test interrupted: Timeout reached (3.01 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/janrichter/avocado/job-results/job-2024-08-28T10.10-f11768c/results.html
JOB TIME   : 4.70 s
```

**System information (please complete the following information):**
 - OS: 
 ```
LSB Version:    :core-5.0-amd64:core-5.0-noarch
Distributor ID: Fedora
Description:    Fedora release 40 (Forty)
Release:        40
Codename:       Forty

```
 - Avocado version: master
 - Avocado installation method: github

**Comment by clebergnu** [comment]
Hi @richtja, thanks for opening this issue.  I think the main point this issue proposes is valid, that is, if `setUp` times out, and gets interrupted, it should have an `INTERRUPTED` status.

BTW, IMO this issue should cover fixing the behavior, but also documenting the proposed timeout behavior.  When issue https://github.com/avocado-framework/avocado/issues/5994 is implemented, then the documentation should be updated to cover the *individual* timeouts for the different phases.

**Comment by richtja** [comment]
> Hi @richtja, thanks for opening this issue. I think the main point this issue proposes is valid, that is, if `setUp` times out, and gets interrupted, it should have an `INTERRUPTED` status.
> 
> BTW, IMO this issue should cover fixing the behavior, but also documenting the proposed timeout behavior. When issue #5994 is implemented, then the documentation should be updated to cover the _individual_ timeouts for the different phases.

IMO the current timeout behaviour is documented by #6012 and if we agree on that any timeout should have `INTERRUPTED` status than I think we don't need to add more documentation to this. Of course, the #5994 should also have documentation part of that new feature.

**Comment by richtja** [comment]
Resolved by #6015
